stixcore/products/levelb/binary.py

```diff
@@ -125,32 +125,56 @@ def __add__(self, other):
         return type(self)(service_type=self.service_type, service_subtype=self.service_subtype,
                           ssid=self.ssid, control=control, data=data)
 
-    def to_days(self):
+    def to_files(self):
         """Split the data into a sequence of LevelB objects each containing 1 daysworth\
-        of observations.
+        of observations for continuous data or by request id in case of BSD.
 
         Yields
         ------
         `LevelB`
-            the next `LevelB` objects for another day.
+            the next `LevelB` objects for another day/request.
 
         """
-        scet = SCETime(coarse=self.control['scet_coarse'], fine=self.control['scet_fine'])
-        days = np.unique(scet.get_scedays(timestamp=True))
-        for day in days:
-            day_start = SCETime(coarse=day, fine=0)
-            day_end = day_start + SEC_IN_DAY * u.s
-            inds = np.argwhere((scet >= day_start) & (scet < day_end))
-            i = np.arange(inds.min(), inds.max()+1)
-            control = self.control[i]
-            control_indices = control['index']
-            min_index = control_indices.min()
-            control['index'] = control['index'] - min_index
-            data = self.data[i]
-            data['control_index'] = data['control_index'] - min_index
-
-            yield type(self)(service_type=self.service_type, service_subtype=self.service_subtype,
-                             ssid=self.ssid, control=control, data=data)
+        if isinstance(self.control['request_id'][0], np.bool_):
+            scet = SCETime(coarse=self.control['scet_coarse'], fine=self.control['scet_fine'])
+            days = np.unique(scet.get_scedays(timestamp=True))
+            for day in days:
+                day_start = SCETime(coarse=day, fine=0)
+                day_end = day_start + SEC_IN_DAY * u.s
+                inds = np.argwhere((scet >= day_start) & (scet < day_end))
+                i = np.arange(inds.min(), inds.max()+1)
+                control = self.control[i]
+                control_indices = control['index']
+                min_index = control_indices.min()
+                control['index'] = control['index'] - min_index
+                data = self.data[i]
+                data['control_index'] = data['control_index'] - min_index
+
+                yield type(self)(service_type=self.service_type,
+                                 service_subtype=self.service_subtype,
+                                 ssid=self.ssid, control=control, data=data)
+        else:
+            rid = []
+            psc = []
```

Comment:
I take it rid is request_id  but what is psc? Maybe slight more verbose variable names would be good